pkg/selinux/setroubleshoot-client.js

```diff
@@ -107,69 +107,56 @@
+                details.pluginAnalysis = details.pluginAnalysis.map(itm => ({
+                    ifText: itm[0],
+                    thenText: itm[1],
+                    doText: itm[2],
+                    analysisId: itm[3],
+                    fixable: itm[4],
+                    reportBug: itm[5],
+                }));
+                return details;
+            })
+            .catch(ex => {
+                console.warn("Unable to get alert for id", localId, ":", JSON.stringify(ex));
+                return Promise.reject(new Error(`Unable to get alert for id ${localId}: ${ex.toString()}`));
+            });
 
     /* Run a fix via SetroubleshootFixit
        The analysisId is given as part of pluginAnalysis entries in alert details
      */
-    client.runFix = function(alertId, analysisId) {
-        return new Promise((resolve, reject) => {
-            client.proxyFixit.call("run_fix", [alertId, analysisId])
-                    .then(result => resolve(result[0]))
-                    .catch(ex => reject(new Error(cockpit.format(_("Unable to run fix: $0"), ex))));
-        });
-    };
+    client.runFix = (alertId, analysisId) => client.proxyFixit.call("run_fix", [alertId, analysisId])
+            .then(result => result[0])
+            .catch(ex => {
+                console.warn("Unable to run fix:", JSON.stringify(ex));
+                return Promise.reject(new Error(cockpit.format(_("Unable to run fix: $0"), ex.toString())));
+            });
 
     /* Delete an alert from the database (will be removed for all users), returns true on success
      * Only assign this to the client variable if the dbus interface actually supports the operation
      */
-    const deleteAlert = function(localId) {
-        return new Promise((resolve, reject) => {
-            client.proxy.call("delete_alert", [localId])
-                    .then(success => {
-                        if (success)
-                            resolve();
-                        else
-                            reject(new Error(cockpit.format(_("Failed to delete alert: $0"), localId)));
-                    })
-                    .catch(ex => {
-                        console.warn("Unable to delete alert with id " + localId);
-                        console.warn(ex);
-                        reject(new Error(cockpit.format(_("Error while deleting alert: $0"), localId)));
-                    });
-        });
-    };
+    const deleteAlert = localId => client.proxy.call("delete_alert", [localId])
+            .then(success => {
+                if (!success)
+                    return Promise.reject(new Error(cockpit.format(_("Failed to delete alert: $0"), localId)));
+            })
+            .catch(ex => {
+                console.warn("Unable to delete alert with id", localId, ":", JSON.stringify(ex));
+                return Promise.reject(new Error(cockpit.format(_("Failed to delete alert: $0"), ex.toString())));
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18670-20230419-114356-a9ad7ebb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)